File: Engr-T.stark/removeNthNodeFromList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* temp = head;
        int len=0;
        while(temp != NULL){
            temp = temp->next;
            len++;
        }
        len = len - n;
        
        ListNode* tem = new ListNode();
        tem->next = head;
        ListNode* prev = tem;
        
        while(len--){
            prev = prev->next;
        }
        prev->next = prev->next->next;
        return tem->next;
    }
};

File: Esosa/Day 44- remove nth node from list/remove nth node from list.py
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        dummy = ListNode(0, head)
        left = dummy
        right = head
            
        while n> 0 and right:
            right= right.next
            n-=1

        while right:
            left= left.next
            right= right.next
        #delete
        left.next= left.next.next
        return dummy.next
